src/proxy/Map.ts

```diff
@@ -56,4 +124,269 @@ export class Map<K, V> extends BaseProxy implements IMap<K, V> {
     isEmpty(): Q.Promise<boolean> {
         return this.encodeInvokeOnRandomTarget<boolean>(MapIsEmptyCodec);
     }
+
+    getAll(keys: K[]): Q.Promise<any[]> {
+        var partitionService = this.client.getPartitionService();
+        var partitionsToKeys: {[id: string]: any} = {};
+        var key: K;
+        for (var i in keys) {
+            key = keys[i];
```

Comment:
There is requests from community to make Typescript support selectively turning on and off certain features: https://github.com/Microsoft/TypeScript/issues/4389 . I hope they support this feature and we can use cool ES6 features that Node already supports.